bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -240,6 +246,35 @@ public BigtableSession(BigtableOptions opts) throws IOException {
     if (options.useGCJClient()) {
       BigtableDataSettings dataSettings =
           BigtableVeneerSettingsFactory.createBigtableDataSettings(options);
+
+      // If we're using a cached channel we need to check if it's set up already. If
+      //  not we need to do the setup now.
+      if (options.useCachedChannel()) {
+        synchronized (BigtableSession.class) {
+          // If it's not set up for this specific Host we set it up and save the context for
+          //  future connections.
+          if (!gcjClientContext.containsKey(options.getDataHost())) {
+            gcjClientContext.put(
+                options.getDataHost(), ClientContext.create(dataSettings.getStubSettings()));
+          }
+        }
+
+        BigtableDataSettings.Builder builder = dataSettings.toBuilder();
+
+        // Add the executor and transport channel to the settings/options
+        builder
+            .stubSettings()
+            .setExecutorProvider(
+                FixedExecutorProvider.create(
+                    gcjClientContext.get(options.getDataHost()).getExecutor()))
+            .setTransportChannelProvider(
+                FixedTransportChannelProvider.create(
+                    Objects.requireNonNull(
+                        gcjClientContext.get(options.getDataHost()).getTransportChannel())))
+            .build();
```

Comment:
InstantiatingChannelProvider uses a reference to the credentials when being constructed:
https://github.com/googleapis/gax-java/blob/master/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java#L235

So I think you have to reuse the cached credentials as well:
```
.setCredentialsProvider(FixedCredentialsProvider.create(cachedClientContext.getCredentials())
```